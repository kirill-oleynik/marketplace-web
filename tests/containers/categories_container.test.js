const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const CategoriesContainer = require('../../app/containers/categories_container').default;

const FakeComponent = () => (
  <div />
);

const fakeStore = (state = {}) => ({
  subscribe() {},
  dispatch() {},
  getState() { return state; }
});

const firstCategory = {
  id: 1,
  title: 'Test1',
  applications: []
};

const secondCategory = {
  id: 2,
  title: 'Test2',
  applications: []
};

const applications = {
  byId: {}
};

const categories = {
  ids: [1, 2],
  byId: {
    2: secondCategory,
    1: firstCategory
  }
};

describe('#render', () => {
  test('it renders correctly', () => {
    const store = fakeStore({
      categories,
      applications
    });

    const WrappedComponent = CategoriesContainer(FakeComponent);
    const component = shallow(
      <WrappedComponent />,
      { context: { store } }
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#passedProps', () => {
  test('it passess props to component correctly', () => {
    const store = fakeStore({
      categories,
      applications
    });

    const WrappedComponent = CategoriesContainer(FakeComponent);
    const component = shallow(
      <WrappedComponent />,
      { context: { store } }
    );

    expect(
      component.find(FakeComponent).props().categories
    ).toEqual([firstCategory, secondCategory]);

    expect(
      component.find(FakeComponent).props().expand
    ).toBeInstanceOf(Function);

    expect(
      component.find(FakeComponent).props().collapse
    ).toBeInstanceOf(Function);
  });
});
