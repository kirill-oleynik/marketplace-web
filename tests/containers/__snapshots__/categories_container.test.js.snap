// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<FakeComponent
  categories={
    Array [
      Object {
        "applications": Array [],
        "id": 1,
        "title": "Test1",
      },
      Object {
        "applications": Array [],
        "id": 2,
        "title": "Test2",
      },
    ]
  }
  collapse={[Function]}
  expand={[Function]}
/>
`;
